samples/benchmarks/coremark/README.rst

```diff
@@ -0,0 +1,155 @@
+If you want to register your results, see the Submitting Results chapter in the `CoreMark GitHub`_ repository.
+
+Additionally, the ``CONFIG_COREMARK_DATA_SIZE`` option requires available memory space.
+By default, the ``CONFIG_COREMARK_MEMORY_METHOD_STACK`` option is selected.
+In this case, the main thread stack is used for data allocation.
+To change the main stack size, use the ``CONFIG_MAIN_STACK_SIZE`` option.
+
+If you select the ``CONFIG_COREMARK_MEMORY_METHOD_MALLOC`` option, Zephyr :c:func:`k_malloc\k_free` functions will be used to allocate the memory in the heap.
+Adjust the heap size with the ``CONFIG_HEAP_MEM_POOL_SIZE`` config option.
+
+If you select the ``CONFIG_COREMARK_MEMORY_METHOD_STATIC`` option, data will be allocated to the RAM as a regular static variable.
+In this case you do not need to be aware of the main thread stack size or heap memory pool size.
+
+CoreMark can be executed in a multithread mode.
+To specify a number of threads, use the ``CONFIG_COREMARK_THREADS_NUMBER`` option.
+You can configure up to four threads.
+In the multithread mode, CoreMark is executed in all threads simultaneously.
+By default ``CONFIG_TIMESLICE_SIZE`` is set to 10 ms which imitates RTOS multithread usage.
+However, in the final report thread execution is combined.
+In multithread mode, the ``CONFIG_COREMARK_DATA_SIZE`` option is allocated for each thread separately.
+Only the ``CONFIG_COREMARK_MEMORY_METHOD_STACK`` and ``CONFIG_COREMARK_MEMORY_METHOD_MALLOC`` memory methods can be used with multiple threads.
+In case of the ``CONFIG_COREMARK_MEMORY_METHOD_STACK`` option, the data for all threads are allocated to the main thread stack.
+
+Building and running
+********************
+
+An extra build flag ``CONFIG_COMPILER_OPT="-O3"`` is applied when running the benchmark to achieve the best CoreMark results.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/benchmark/coremark
+   :board: nrf5340dk_nrf5340_cpuapp
+   :goals: build flash
+   :compact:
+
+After flashing, messages will appear in the corresponding console that describe the state of the benchmark.
+
+Testing
+=======
+
+After programming the sample to your development kit, complete the following steps to test it:
+
+1. Connect to the development kit with a terminal emulator (for example, PuTTY).
+#. Reset your development kit.
+#. To start the test, press the button assigned to the respective core.
+   To see the assignment, refer to the User interface section.
+   All target cores work independently from each other.
+   It is possible to run the benchmark on several cores simultaneously.
+   Measurements running on a core will be indicated by the corresponding LED.
+   If the ``COREMARK_MODE_FLASH_AND_RUN`` option is enabled, the measurement will launch on the application's start automatically on all available cores.
+#. Wait for all measurements to be complete.
+   By default test will take approximately 11-13 seconds.
+#. Wait for the console output for both cores.
```

Comment:
Is the statement "both cores" applicable to all scenarios?

Reply:
Changed to "Wait for the console output for all tested cores."